GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -408,10 +410,28 @@ public void OnFilePreDelete(string relativePath)
             this.backgroundFileSystemTaskRunner.Enqueue(FileSystemTask.OnFilePreDelete(relativePath));
         }
 
-        public void OnFolderCreated(string relativePath)
+        /// <summary>
+        /// Called to indicate a folder was created
+        /// </summary>
+        /// <param name="relativePath">The relative path to the newly created folder</param>
+        /// <returns>
+        /// true when the folder is successfully added to the include list because it is in the projection but currently excluded.
+        /// false when the folder was not excluded or there was a failure adding to the included list.
+        /// </returns>
+        public bool OnFolderCreated(string relativePath)
         {
+            GitIndexProjection.PathProjectionState pathProjectionState = this.GitIndexProjection.GetPathProjectionState(relativePath);
+            if (pathProjectionState == GitIndexProjection.PathProjectionState.Excluded)
+            {
+                if (this.GitIndexProjection.TryAddIncludedFolder(relativePath))

```

Comment:
Would you expect `TryAddIncludedFolder` to fail under normal circumstances?

I'm wondering if we should log a warning/error either here or in `TryAddIncludedFolder` if the add fails.

Reply:
>Would you expect TryAddIncludedFolder to fail under normal circumstances?

No

>if we should log a warning/error either here or in TryAddIncludedFolder if the add fails.

Will do.